data/helpers/router.js

```diff
@@ -0,0 +1,103 @@
+const express = require("express");
+const router = express.Router();
+const actions = require("./actionModel");
+const projects = require("./projectModel");
+
+router.get("/", (req, res) => {
+    projects.get()
+        .then(data => res.json(data))
+        .catch(res => res.status(500).json({message: "oops"}))
+} )
+
+router.get("/:id", (req, res) => {
+    const { id } = req.params;
+    projects.get(id)
+        .then(data => {
+            if(!data.id) {
+                res.status(404).json({error: "The project with this id does not exist"})
+            }
+            else {
+                res.json(data);
+            }
+        })
+        .catch(error => res.status(500).json({ message: "oh nos!!!"}))
+})
+
+router.post("/", (req, res) => {
+    const newProject = req.body;
+    projects.insert(newProject)
+        .then(data => {
+            if (!newProject.name || !newProject.description || !newProject.completed) {
+                res.status(404).json({ error: "Please fill out the required fields"})
+            } else {
+               res.json(newProject)}   
+            }) 
+           
+        .catch(err => res.json({ message: "Oh dear...it didn't work"}))
+})
+
+router.put("/:id", (req, res) => {
+    const { id } = req.params;
+    const changes = req.body;
+
+    projects.update( id, changes )
+        .then(data => {
+            if (!data.id){
+                res.status(404).json({error: "Why would you try to update something that doesn't exist?!?!"})
+            } else {
+              res.json(changes);                
+            }
+
+        })
+        .catch(err => res.json({ message: "ruhh-roohhh"}))
+})
+
+router.delete("/:id", (req, res) => {
+    const { id } = req.params;
+    projects.remove(id)
+        .then(data => {
+            res.json(data);
+        })
+        .catch(err => {res.json({ message: "too amazing to be removed"})})
+})
+
+router.get("/:id/actions", (req, res) => {
+    const { id } = req.params;
+    projects.getProjectActions(id)
+        .then(data => {
+            if (data.length > 0) {
+                res.json(data)
+            } else {
+                res.status(404).json({ message: "non-existent"})
+            }
+        
+        })
+        .catch(error => {res.json({ message: "These aren't the actions you are looking for."})})
+})
+
```

Comment:
I would have like to have seen 2 files of endpoints, one for the projects, and one for the actions.